core/translate/emitter.rs

```diff
@@ -242,6 +247,14 @@ pub fn emit_query<'a>(
             target_pc: after_main_loop_label,
         });
     }
+    if !plan.aggregates.is_empty()
+        && plan.group_by.is_none()
+        && plan.result_columns.iter().any(|c| !c.contains_aggregates)
+    {
+        let flag = program.alloc_register();
+        program.emit_int(0, flag);
+        t_ctx.reg_nonagg_emit_once_flag = Some(flag);
+    }
```

Comment:
Can we have comment here about the behaviour? I know it's in the `reg_nonagg_emit_once_flag` definition but it could be good to know here too.